src/protocolgame.cpp

```diff
@@ -1446,9 +1448,44 @@ void ProtocolGame::parseEditVip(NetworkMessage& msg)
 	std::string description = msg.getString();
 	uint32_t icon = std::min<uint32_t>(10, msg.get<uint32_t>()); // 10 is max icon in 9.63
 	bool notify = msg.getByte() != 0;
-	addGameTask([=, playerID = player->getID(), description = std::move(description)]() {
-		g_game.playerRequestEditVip(playerID, guid, description, icon, notify);
-	});
+	uint8_t groups = msg.getByte();
+	std::vector<uint16_t> groupIds;
+	groupIds.reserve(groups);
+	for (uint8_t i = 0; i < groups; i++) {
+		groupIds.push_back(msg.getByte());
+	}
+	addGameTask(
+	    [=, playerID = player->getID(), description = std::move(description), groupIds = std::move(groupIds)]() {
+		    g_game.playerRequestEditVip(playerID, guid, description, icon, notify, groupIds);
+	    });
+}
+
+void ProtocolGame::parseVipGroupAction(NetworkMessage& msg)
+{
+	uint8_t action = msg.getByte();
+
+	uint8_t id;
+	std::string name;
+
+	switch (action) {
+		case VIPGROUPACTION_CREATE:
+			name = msg.getString();
+			addGameTask([=, playerID = player->getID(), name = std::move(name)]() {
+				g_game.playerRequestAddVipGroup(playerID, name);
+			});
+			break;
+		case VIPGROUPACTION_EDIT:
+			id = msg.getByte();
+			name = msg.getString();
+			addGameTask([=, playerID = player->getID(), name = std::move(name)]() {
+				g_game.playerRequestEditVipGroup(playerID, id, name);
+			});
+			break;
+		case VIPGROUPACTION_REMOVE:
+			id = msg.getByte();
+			addGameTask([=, playerID = player->getID()]() { g_game.playerRequestRemoveVipGroup(playerID, id); });
```

Comment:
```suggestion

    switch (action) {
        case VIPGROUPACTION_CREATE:
            std::string name = msg.getString();
            addGameTask([=, playerID = player->getID(), name = std::move(name)]() {
                g_game.playerRequestAddVipGroup(playerID, name);
            });
            break;

        case VIPGROUPACTION_EDIT:
            uint8_t id = msg.getByte();
            std::string name = msg.getString();
            addGameTask([=, playerID = player->getID(), name = std::move(name)]() {
                g_game.playerRequestEditVipGroup(playerID, id, name);
            });
            break;

        case VIPGROUPACTION_REMOVE:
            uint8_t id = msg.getByte();
            addGameTask([=, playerID = player->getID()]() {
                g_game.playerRequestRemoveVipGroup(playerID, id);
            });
```

Reply:
You may need to add brackets `{}` in cases though